File: Anime Game Remap (for all users)/api/src/FixRaidenBoss2/model/textures/ColourRange.py
##### Credits

# ===== Anime Game Remap (AG Remap) =====
# Authors: NK#1321, Albert Gold#2696
#
# if you used it to remap your mods pls give credit for "Nhok0169" and "Albert Gold#2696"
# Special Thanks:
#   nguen#2011 (for support)
#   SilentNightSound#7430 (for internal knowdege so wrote the blendCorrection code)
#   HazrateGolabi#1364 (for being awesome, and improving the code)

##### EndCredits

##### LocalImports
from .Colour import Colour
##### EndLocalImports


##### Script
class ColourRange():
    """
    Class to store data for a colour

    Parameters
    ----------
    min: :class:`Colour`
        The minimum range for the RGBA values

    max: :class:`Colour`
        The maximum range for the RGBA values
    """

    def __init__(self, min: Colour, max: Colour):
        self.min = min
        self.max = max
    
    def match(self, colour: Colour) -> bool:
        """
        Whether 'colour' is within the colour range

        Parameters
        ----------
        colour: :class:`Colour`
            The colour to check

        Returns
        -------
        :class:`bool`
            Whether the colour is within the colour range
        """
        
        return (self.min.red <= colour.red and colour.red <= self.max.red and
                self.min.green <= colour.green and colour.green <= self.max.green and
                self.min.blue <= colour.blue and colour.blue <= self.max.blue and
                self.min.alpha <= colour.alpha and colour.alpha <= self.max.alpha)
##### EndScript

File: Anime Game Remap (for all users)/api/src/cpp/core/src/model/textures/ColourRange.cpp
#include "AGRemapCore/model/textures/ColourRange.h"

namespace AGRemapCore {

    ColourRange::ColourRange(Colour min, Colour max): min(min), max(max) {}

    std::size_t ColourRange::hash() const {
        return std::hash<std::string>{}(getId());
    }

    std::string ColourRange::getId() const {
        return min.getId() + max.getId();
    }

    bool ColourRange::match(const Colour &colour) const {
        return min.red <= colour.red && colour.red <= max.red &&
               min.green <= colour.green && colour.green <= max.green &&
               min.blue <= colour.blue && colour.blue <= max.blue &&
               min.alpha <= colour.alpha && colour.alpha <= max.alpha;
    }

    bool matchesColourOrRange(const ColourOrRange &colourOrRange, const Colour &colour) {
        if (std::holds_alternative<Colour>(colourOrRange)) {
            return std::get<Colour>(colourOrRange).match(colour);
        }
        return std::get<ColourRange>(colourOrRange).match(colour);
    }
}
